pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -998,6 +1001,30 @@ func getEnvoyRouteTypeAndVal(r *route.Route) (envoyRouteType, string) {
 	return iType, iVal
 }
 
+func catchAllMatch(http *networking.HTTPRoute) *networking.HTTPMatchRequest {
+	// A Match is catch all if and only if it has no header/query param match
+	// and URI has a prefix / or regex *.
+	for _, match := range http.Match {
+		catchalluri := false
+		if match.Uri != nil {
+			switch m := match.Uri.MatchType.(type) {
+			case *networking.StringMatch_Prefix:
+				if m.Prefix == "/" {
+					catchalluri = true
+				}
+			case *networking.StringMatch_Regex:
+				if m.Regex == "*" {
```

Comment:
Is `*` correct here? Doesn't that match only a single character? "The entire path (without the query string) must match the regex."

Reply:
I have just used the same existing logic as this https://github.com/istio/istio/blob/master/pilot/pkg/networking/core/v1alpha3/route/route.go#L993 since it has been working for so long.

Reply:
Not sure if it was done like that for a reason. but If you think we should revisit this, I can definitely do.

Reply:
I was under the assumption that * would match *anything*. I am not much of an expert in ECMAScript. If this is wrong, let us fix igt.